examples/vision/nnclr.py

```diff
@@ -0,0 +1,642 @@
+metrics can be used for hyperparameter tuning even in the case when there are no labeled
+examples.
+- Linear probing accuracy: Linear probing is a popular metric to evaluate self-supervised
+classifiers. It is computed as the accuracy of a logistic regression classifier trained
+on top of the encoder's features. In our case, this is done by training a single dense
+layer on top of the frozen encoder. Note that contrary to traditional approach where the
+classifier is trained after the pretraining phase, in this example we train it during
+pretraining. This might slightly decrease its accuracy, but that way we can monitor its
+value during training, which helps with experimentation and debugging.
+"""
+
+model = NNCLR(temperature=temperature, queue_size=queue_size)
+model.compile(
+    contrastive_optimizer=keras.optimizers.Adam(),
+    probe_optimizer=keras.optimizers.Adam(),
+)
+pretrain_history = model.fit(
+    train_dataset, epochs=num_epochs, validation_data=test_dataset
+)
+
+"""
+## Evaluate our model
+
+A popular way to evaluate a SSL method in computer vision or for that fact any other
+pre-training method as such is to learn a linear classifier on the frozen features of the
+trained backbone model and evaluate the classifier on unseen images. Other methods often
+include fine-tuning on the source dataset or even a target dataset with 5% or 10% labels
+present. You can use the backbone we just trained for any downstream task such as image
+classification (like we do here) or segmentation or detection, where the backbone models
+are usually pre-trained with supervised learning.
+"""
+
+finetuning_model = keras.Sequential(
+    [
+        layers.Input(shape=input_shape),
+        augmenter(**classification_augmenter),
+        model.encoder,
+        layers.Dense(10),
+    ],
+    name="finetuning_model",
+)
+finetuning_model.compile(
+    optimizer=keras.optimizers.Adam(),
+    loss=keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+    metrics=[keras.metrics.SparseCategoricalAccuracy(name="acc")],
+)
+
+finetuning_history = finetuning_model.fit(
+    labeled_train_dataset, epochs=num_epochs, validation_data=test_dataset
+)
+
+"""
+Self supervised learning is particularly very helpful when you do not have access to very
+limited labeled training data but you can manage to build a large corpus of unlabeled
+data as shown by previous methods like [SEER](https://arxiv.org/abs/2103.01988),
+[SimCLR](https://arxiv.org/abs/2002.05709), [SwAV](https://arxiv.org/abs/2006.09882) and
+more.
+
+You should also take a look at the blogs for these papers which neatly show that it is
+possible to achieve good results with few class labels by first pretraining on a large
+unlabeled dataset and then fine-tuning on a smaller labeled dataset:
+- [Advancing Self-Supervised and Semi-Supervised Learning with
+SimCLR](https://ai.googleblog.com/2020/04/advancing-self-supervised-and-semi.html)
```

Comment:
Make sure that all links are on a single line (here and below)

Reply:
Got it